main/main.go

```diff
@@ -71,12 +74,16 @@ func runCmd() error {
 		}
 	}
 
-	// Stop early, is the dry run (-dry-run) flag was given.
-	if *flagDryRun {
+	// Stop early if the dry run (-dry-run) flag was given without an output-dir.
+	if *flagDryRun && *flagOutputDir == "" {
 		log.Printf("skipping GCS upload since dry run was specified")
 		return nil
```

Comment:
I do not think this message and the check align. It would be appropriate to log output directory related message which is being done below. Similarly for GCS bucket upload case `Uploading CVE data to GCS bucket {bucket-name}`.